app/models/search.rb

```diff
@@ -15,19 +16,34 @@ class Search
   validate :dont_search_the_past
 
   def date_value
-    begin
-      Date.strptime(self.date, '%d/%m/%Y')
-    rescue
-      Date.today
-    end
+    # dans le @search : '%d/%m/%Y'
+    # date en texte de type : "vendredi 13 mai 2011" : 4 chiffres -> l'année ; 1 ou 2 chiffres -> le jour ; le mois en lettres est dans I18n.t('date.month_names'), .index("mois en lettres") donne le mois en chiffres
+    #begin
+    #Date.strptime(self.date, '%d/%m/%Y')
+    the_match_object = /([A-Za-z]+) ([0-9]{1,2}) ([A-Za-z]+) ([0-9]{4})/.match(self.date)
+    the_day_in_digits = Integer(the_match_object[2])
+    the_month_in_text = the_match_object[3].downcase
+    the_year_in_digits = Integer(the_match_object[4])
+    the_month_in_digits = I18n.t('date.month_names').index(the_month_in_text)
+    Date.new(the_year_in_digits, the_month_in_digits, the_day_in_digits)
+    #rescue
+      #Date.today
+    #end
   end
 
   private
 
     def dont_search_the_past
-      @date = Date.today.strftime('%d/%m/%Y') if @date.blank?
+      @date = I18n.localize(Date.today, :format => '%A %d %B %Y') if @date.blank?
       if self.date_value < Date.today
-        errors.add(:date, "Cherchez une date future")
+        #errors.add(:date, "Cherchez une date future")
+        @date = I18n.localize(Date.today, :format => '%A %d %B %Y')
+      end
+      if (not @from_city.blank?) and @from_lon.blank?
```

Comment:
This check should be done in another validator as it is not related to `dont_search_the_past`.